### DISCUSSION TITLE
LockMode

### DISCUSSION THREAD

**Discussion by CyberCyclone** [ravendb/ravendb#14792] [Q&A] [answered]
I've just been playing around with an index lock mode. I've set it to 'LockedError' and I was expected it to be throw an error if I tried to delete the index. From what I can see the lock mode only applies to modifying the index. Is there a reason it doesn't apply to a 'delete' command and is there a way to protect an index from accidental deletion?

**Comment by garayx** [comment, marked as answer]
Thanks for the suggestion we will look into it, please follow this issue:

https://issues.hibernatingrhinos.com/issue/RavenDB-19239/Prevent-index-deletion-when-index-lock-mode-is-locked
